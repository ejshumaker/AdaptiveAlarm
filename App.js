import React from 'react';
import { View } from 'react-native';

// Import custom modules below npm packages & libraries
import AppContainer from './src/navigation/AppContainer';
import { GlobalStyles } from './src/constants';


/**
 * App is the single entry point of the application,
 * this file should not be modified for basic functionality,
 * what happens in this file affects the entire app
 * @eschirtz 03-01-19
 */
class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <View style={GlobalStyles.container}>
        <AppContainer />
      </View>
    );
  }
}

export default App;
