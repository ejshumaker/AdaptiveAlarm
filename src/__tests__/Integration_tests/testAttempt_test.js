
/* eslint-disable no-undef */
/* eslint-disable import/first */
jest.mock('react-native-gesture-handler', () => {
  const View = require('react-native/Libraries/Components/View/View');
  return {
    Swipeable: View,
    DrawerLayout: View,
    State: {},
    ScrollView: View,
    Slider: View,
    Switch: View,
    TextInput: View,
    ToolbarAndroid: View,
    ViewPagerAndroid: View,
    DrawerLayoutAndroid: View,
    WebView: View,
    NativeViewGestureHandler: View,
    TapGestureHandler: View,
    FlingGestureHandler: View,
    ForceTouchGestureHandler: View,
    LongPressGestureHandler: View,
    PanGestureHandler: View,
    PinchGestureHandler: View,
    RotationGestureHandler: View,
    /* Buttons */
    RawButton: View,
    BaseButton: View,
    RectButton: View,
    BorderlessButton: View,
    /* Other */
    FlatList: View,
    gestureHandlerRootHOC: jest.fn(),
    Directions: {},
  };
});

jest.mock('../../assets/sounds/', () => jest.fn());


import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import App from '../../../App';


Enzyme.configure({ adapter: new Adapter() });
describe('Account Screen', () => {
  let wrapper;
  // our mock login function to replace the one provided by mapDispatchToProps
  const mockSignOutfn = jest.fn();
  const navigation = { navigate: jest.fn() };

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('test account screen matches snapshot done loading', () => {
    wrapper.setState({ fontLoaded: true });
    wrapper = wrapper.find('NavigationContainer').shallow();
    expect(wrapper).toMatchSnapshot();
  });
});
