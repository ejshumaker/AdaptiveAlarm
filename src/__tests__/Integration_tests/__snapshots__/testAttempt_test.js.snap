// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Screen test account screen matches snapshot done loading 1`] = `
<ContextProvider
  value={
    Object {
      "_childrenNavigation": Object {},
      "actions": Object {},
      "addListener": [Function],
      "dangerouslyGetParent": [Function],
      "dispatch": [Function],
      "getChildNavigation": [Function],
      "getScreenProps": [Function],
      "goBack": [Function],
      "isFocused": [Function],
      "navigate": [Function],
      "router": Object {
        "childRouters": Object {
          "App": Object {
            "childRouters": Object {
              "Account": null,
              "Alarm": null,
              "AlarmList": null,
              "AutoComplete": null,
              "Calendar": null,
              "CreateAlarm": null,
              "DayPicker": null,
              "Home": null,
              "Main": null,
              "StyleDemo": null,
            },
            "getActionCreators": [Function],
            "getActionForPathAndParams": [Function],
            "getComponentForRouteName": [Function],
            "getComponentForState": [Function],
            "getPathAndParamsForState": [Function],
            "getScreenOptions": [Function],
            "getStateForAction": [Function],
          },
          "Auth": Object {
            "childRouters": Object {
              "SignIn": null,
              "SignUp": null,
            },
            "getActionCreators": [Function],
            "getActionForPathAndParams": [Function],
            "getComponentForRouteName": [Function],
            "getComponentForState": [Function],
            "getPathAndParamsForState": [Function],
            "getScreenOptions": [Function],
            "getStateForAction": [Function],
          },
          "AuthLoading": Object {
            "childRouters": Object {
              "AuthLoading": null,
            },
            "getActionCreators": [Function],
            "getActionForPathAndParams": [Function],
            "getComponentForRouteName": [Function],
            "getComponentForState": [Function],
            "getPathAndParamsForState": [Function],
            "getScreenOptions": [Function],
            "getStateForAction": [Function],
          },
        },
        "getActionCreators": [Function],
        "getActionForPathAndParams": [Function],
        "getComponentForRouteName": [Function],
        "getComponentForState": [Function],
        "getPathAndParamsForState": [Function],
        "getScreenOptions": [Function],
        "getStateForAction": [Function],
      },
      "setParams": [Function],
      "state": Object {
        "index": 0,
        "isTransitioning": false,
        "routes": Array [
          Object {
            "index": 0,
            "isTransitioning": false,
            "key": "AuthLoading",
            "params": undefined,
            "routeName": "AuthLoading",
            "routes": Array [
              Object {
                "key": "id-1556116991591-0",
                "routeName": "AuthLoading",
              },
            ],
          },
          Object {
            "index": 0,
            "isTransitioning": false,
            "key": "Auth",
            "params": undefined,
            "routeName": "Auth",
            "routes": Array [
              Object {
                "key": "id-1556116991591-1",
                "routeName": "SignIn",
              },
            ],
          },
          Object {
            "index": 0,
            "isTransitioning": false,
            "key": "App",
            "params": undefined,
            "routeName": "App",
            "routes": Array [
              Object {
                "key": "id-1556116991591-2",
                "routeName": "Main",
              },
            ],
          },
        ],
      },
    }
  }
>
  <Navigator
    navigation={
      Object {
        "_childrenNavigation": Object {},
        "actions": Object {},
        "addListener": [Function],
        "dangerouslyGetParent": [Function],
        "dispatch": [Function],
        "getChildNavigation": [Function],
        "getScreenProps": [Function],
        "goBack": [Function],
        "isFocused": [Function],
        "navigate": [Function],
        "router": Object {
          "childRouters": Object {
            "App": Object {
              "childRouters": Object {
                "Account": null,
                "Alarm": null,
                "AlarmList": null,
                "AutoComplete": null,
                "Calendar": null,
                "CreateAlarm": null,
                "DayPicker": null,
                "Home": null,
                "Main": null,
                "StyleDemo": null,
              },
              "getActionCreators": [Function],
              "getActionForPathAndParams": [Function],
              "getComponentForRouteName": [Function],
              "getComponentForState": [Function],
              "getPathAndParamsForState": [Function],
              "getScreenOptions": [Function],
              "getStateForAction": [Function],
            },
            "Auth": Object {
              "childRouters": Object {
                "SignIn": null,
                "SignUp": null,
              },
              "getActionCreators": [Function],
              "getActionForPathAndParams": [Function],
              "getComponentForRouteName": [Function],
              "getComponentForState": [Function],
              "getPathAndParamsForState": [Function],
              "getScreenOptions": [Function],
              "getStateForAction": [Function],
            },
            "AuthLoading": Object {
              "childRouters": Object {
                "AuthLoading": null,
              },
              "getActionCreators": [Function],
              "getActionForPathAndParams": [Function],
              "getComponentForRouteName": [Function],
              "getComponentForState": [Function],
              "getPathAndParamsForState": [Function],
              "getScreenOptions": [Function],
              "getStateForAction": [Function],
            },
          },
          "getActionCreators": [Function],
          "getActionForPathAndParams": [Function],
          "getComponentForRouteName": [Function],
          "getComponentForState": [Function],
          "getPathAndParamsForState": [Function],
          "getScreenOptions": [Function],
          "getStateForAction": [Function],
        },
        "setParams": [Function],
        "state": Object {
          "index": 0,
          "isTransitioning": false,
          "routes": Array [
            Object {
              "index": 0,
              "isTransitioning": false,
              "key": "AuthLoading",
              "params": undefined,
              "routeName": "AuthLoading",
              "routes": Array [
                Object {
                  "key": "id-1556116991591-0",
                  "routeName": "AuthLoading",
                },
              ],
            },
            Object {
              "index": 0,
              "isTransitioning": false,
              "key": "Auth",
              "params": undefined,
              "routeName": "Auth",
              "routes": Array [
                Object {
                  "key": "id-1556116991591-1",
                  "routeName": "SignIn",
                },
              ],
            },
            Object {
              "index": 0,
              "isTransitioning": false,
              "key": "App",
              "params": undefined,
              "routeName": "App",
              "routes": Array [
                Object {
                  "key": "id-1556116991591-2",
                  "routeName": "Main",
                },
              ],
            },
          ],
        },
      }
    }
  />
</ContextProvider>
`;
